### PR TITLE
Follow up to astropy 3.1.1 unit bug fix

### LINKED ISSUE #587: astropy.units.core.UnitConversionError: 's / m' and 's / m' are not convertible
https://travis-ci.org/PlasmaPy/PlasmaPy/jobs/466396211

After wrestling with system packages (`conda`, `pip` and `pacman`, argh!), I've managed to figure out that this is localized to `astropy` 3.1. Also, running `python setup.py test -d` and getting into `plasmapy.physics.distribution.Maxwellian_1D`:

```python
(Pdb) distFunc
<Quantity 5.91632969e-07 s / m>
(Pdb) p u.s / u.m
Unit("s / m")
(Pdb) p distFunc
<Quantity 5.91632969e-07 s / m>
(Pdb) p distFunc.unit
Unit("s / m")
(Pdb) p distFunc.to(u.s / u.m)
*** astropy.units.core.UnitConversionError: 's / m' and 's / m' are not convertible
```

### LINKED ISSUE #589: Add more comprehensive testing
Would you consider adding more jobs to the test matrices? E.g. test against more numpy versions you support, or astropy versions.

Also, preferably also against the dev versions of these to issues like #587 would surface much earlier, still in the development cycle.

### PR DESCRIPTION
**Pull request by StanczakDominik**
I guess f8fac842ea1e074aba60d93f91fbdecf153987bf was broken in that it provided two mentions of astropy and this is causing conflicts.

Closes #589 and closes #587 too.

### CODE REVIEW

**Comment by StanczakDominik** [comment]
Some lingering issues with dev versions of numpy, which looks a bit like something on the conda side...

**Comment by StanczakDominik** [comment]
Bloody hell. This is exactly what we get for not dealing with the astropy unit bug earlier and pushing to master on broken tests. My bad, apologies >_<

What happened here is that I was so sure #586 was fine since tests were failing but that PR did nothing related to the astropy 3.1.0 unit optimization bug, so it was fine to merge it. It wasn't! That pull request triggered the `__all__` glitch. Basically, importing stuff from `typing` - `Callable` and the like in file `example.py` puts them into `example`'s, namespace, right? Well, that's bad news because then sphinx with automodapi, when building these, inspects the `__all__` parameter which now contains `Callable` and tries to print out their definitions. But those can't be found in the file, so it returns warnings such as:
```

b'<partial node>':: WARNING: toctree contains reference to nonexisting document 'api/plasmapy.utils.pytest_helpers.Dict'
b'<partial node>':: WARNING: toctree contains reference to nonexisting document 'api/plasmapy.utils.pytest_helpers.Callable'
b'<partial node>':: WARNING: toctree contains reference to nonexisting document 'api/plasmapy.utils.pytest_helpers.Dict'
b'<partial node>':: WARNING: toctree contains reference to nonexisting document 'api/plasmapy.utils.pytest_helpers.Callable'
b'<partial node>':: WARNING: toctree contains reference to nonexisting document 'api/plasmapy.utils.pytest_helpers.Dict'
```

To sum up, when I was in high school I used to play Command and Conquer 3 online a lot, in team games. In fact, that's where I picked up most of my english. There was this guy nicknamed Ulkrond whom me and my group met up with often, whose most memorable quote that we had many a laugh about "Your hubris shall be your downfall".

And, well, turns out that indeed it was.

**Comment by namurphy** [comment]
The good thing about version control and open development is that it makes it more straightforward to fix things like this when issues arise.  Thank you for making these fixes!  

And as I like to say, getting things wrong is the first step towards getting things right!

File: plasmapy/utils/pytest_helpers.py [hunk 1]
```diff
@@ -25,6 +25,22 @@
...
+    "MissingExceptionError",
+    "MissingWarningError",
+    "IncorrectResultError",
+    "InvalidTestError",
+    "call_string",
+    "run_test",
+    "run_test_equivalent_calls",
+    "assert_can_handle_nparray",
+]
+
```
Comment (namurphy) [inline plasmapy/utils/pytest_helpers.py]: Yes, this definitely is a better fix than what I was trying to do in #592.
